internal/controller/atlasschema_controller.go

```diff
@@ -553,12 +612,48 @@ func (r *AtlasSchemaReconciler) recordErrEvent(res *dbv1alpha1.AtlasSchema, err
 }
 
 // ShouldLint returns true if the linting policy is set to error.
-func (d *managedData) shouldLint() bool {
+func (d *managedData) shouldLint() (bool, error) {
 	p := d.Policy
 	if p == nil || p.Lint == nil || p.Lint.Destructive == nil {
+		// Check if the destructive policy is set to error in the custom config.
+		// This check won't work if the attribute value has expressions in it.
+		if d.Config != nil {
+			env := searchBlock(d.Config.Body(), hclwrite.NewBlock("env", []string{d.EnvName}))
+			if env == nil {
+				return false, nil
+			}
+			lint := searchBlock(env.Body(), hclwrite.NewBlock("lint", nil))
+			if lint == nil {
+				return false, nil
+			}
```

Comment:
The `lint` config can be provide via global `lint` block, or in the `env` block. This check missing the global case.